vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SequenceManager.java

```diff
@@ -0,0 +1,66 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.SequenceRepository;
+import edu.asu.diging.vspace.core.data.SlideRepository;
+import edu.asu.diging.vspace.core.factory.ISequenceFactory;
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.Sequence;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.web.staff.forms.SequenceForm;
+
+@Transactional
+@Service
+public class SequenceManager implements ISequenceManager {
+
+    @Autowired
+    private ModuleManager moduleManager;
+
+    @Autowired
+    private ISequenceFactory sequenceFactory;
+
+    @Autowired
+    private SequenceRepository sequenceRepo;
+
+    @Autowired
+    private SlideRepository slideRepo;
+
+    @Override
+    public ISequence storeSequence(String moduleId, SequenceForm sequenceForm) {
+        String[] orderedSlideIds = sequenceForm.getOrderedSlideIds().split(",");
+        List<String> slideIds = Arrays.asList(orderedSlideIds);
+        Map<String, Integer> idOrderMap = new HashMap<>();
+        for (int i = 0; i < orderedSlideIds.length; ++i) {
+            idOrderMap.put(orderedSlideIds[i], i);
+        }
```

Comment:
why do you need a map here? you could probably just use indexOf of the list in line 52.